atst/routes/task_orders/new.py

```
@@ -176,12 +189,8 @@ def cancel_edit(task_order_id=None, portfolio_id=None):
         else:
             form = TaskOrderForm(form_data)
 
-        if form.validate(flash_invalid=False):
-            task_order = None
-            if task_order_id:
-                task_order = TaskOrders.update(task_order_id, **form.data)
-            else:
-                task_order = TaskOrders.create(portfolio_id, **form.data)
+        update_task_order(form, portfolio_id, task_order_id, flash_invalid=False)
```

Did we decide to just swallow the error if they enter a redundant TO number and not save their data?

---

I talked it over with Ben, and what we decided to do was to show the flash error on the TO index page, and save all of the users changes except the dupe TO number.  `flash_invalid=False` here prevents the flash error from the BaseForm validator from showing up in the case of other update errors. https://github.com/dod-ccpo/atst/blob/2edfecdd4581586cfccc30545bac2b72248b6669/atst/forms/forms.py#L45